lib/store/settings_store.dart

```diff
@@ -1484,16 +1483,40 @@ abstract class SettingsStoreBase with Store {
     powNodes[walletType] = node;
   }
 
-  void initializeTrocadorProviderStates() {
-    for (var provider in TrocadorExchangeProvider.availableProviders) {
-      final savedState = _sharedPreferences.getBool(provider) ?? true;
-      trocadorProviderStates[provider] = savedState;
+  @action
+  Future<void> updateAllTrocadorProviderStates(List<String> availableProviders) async {
+    String? serializedData = await _sharedPreferences.getString(PreferencesKey.trocadorProviderStatesKey);
+    Map<String, bool> regularMap = {};
+
+    if (serializedData != null) {
+      var decodedData = json.decode(serializedData);
+      if (decodedData is Map) {
+        regularMap = decodedData.map<String, bool>((key, value) => MapEntry(key as String, value as bool));
+      }
```

Comment:
**Enhancement:**
no need for all this just
`json.decode(response.body) as Map<String, bool>` should be enough, but test it as well

Reply:
and you can do that for other places you have this flow

Reply:
done